troposphere/static/js/components/modals/instance/InstanceLaunchWizardModal.jsx

```diff
@@ -315,20 +315,18 @@ export default React.createClass({
         let identityProvider;
         if (providerList) {
             provider = providerList.first();
-            providerSizeList = stores.SizeStore.fetchWhere({
-                provider__id: provider.id
-            });
-
+        }
+        if (provider) {
+            providerSizeList = this.getSizeList(provider, imageVersion);
             identityProvider = stores.IdentityStore.findOne({
                 "provider.id": provider.id
             });
-
-            if (providerSizeList) {
-                providerSize = providerSizeList.first();
-            }
-            ;
         }
 
+        if (providerSizeList) {
+            providerSize = providerSizeList.first();
+        } // Else null?
```

Comment:
So one one hand we're assuming that providerSizeList has at least one. Also we're ignoring whether the user has selected a providerSize before. Things to keep in mind.. but not worth addressing here.